episodes/import-and-visualise-osm-data.Rmd

```diff
@@ -0,0 +1,226 @@
+
+::::::::::::::::::::::::::::::::::::: objectives
+
+- Import OSM vector data from the API
+- Select and manipulate OSM vector data
+- Visualise and map OSM Vector data
+- Use Leaflet for interactive mapping
+
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+## Introduction: What is Open Street Map?
+
+Open Street Map (OSM) is a collaborative project which aims at mapping the world and sharing geospatial data in an open way. Anyone can contribute, by mapping geographical objects their encounter, by adding topical information on existing map objects (their name, function, capacity, etc.), or by mapping buildings and roads from satellite imagery (cf. [HOT: Humanitarian OpenStreetMap Team](https://www.hotosm.org/)).
+
+This information is then validated by other users and eventually added to the common "map" or information system. This ensures that the information is accessible, open, verified, accurate and up-to-date.
+
+The result looks like this:
+![View of OSM web interface](https://raw.githubusercontent.com/ClementineCttn/r-geospatial-urban/blob/main/episodes/fig/OSM1.png)
+
+The geospatial data underlying this interface is made of geometrical objects (i.e. points, lines, polygons) and their associated tags (#building #height, #road #secondary #90kph, etc.).
+
+## How to extract geospatial data from Open Street Map?
+
+```{r}
+library(tidyverse)
+library(sf)
+library(leaflet)
+library(osmdata)
+library(nominatimlite)
+```
+
+
+### Bounding-box
+
+The first thing to do is to define the area within which you want to retrieve data, aka the *bounding box*. This can be defined easily using a place name and the package `nominatimlite` to access the free Nominatim API provided by OpenStreetMap. 
+
+We are going to look at *Brielle* together, but you can also work with the small cities of *Naarden*, *Geertruidenberg*, *Gorinchem*, *Enkhuizen* or *Dokkum*.
+
+N.B. This might not be needed, but ensures that your machine knows it has internet...
+
+```{r}
+assign("has_internet_via_proxy", TRUE, environment(curl::has_internet))
+```
+
+
+
+
+```{r}
+nominatim_polygon <- nominatimlite::geo_lite_sf(address = "Brielle", points_only = FALSE)
+bb <- sf::st_bbox(nominatim_polygon)
+bb
+```
+
+
+### Word of caution
+
+There might multiple responses from the API query, corresponding to different objects at the same location, or different objects at different locations.
+For example: Brielle (Netherlands) and Brielle (New Jersey)
+
+![Brielle, Netherlands](https://raw.githubusercontent.com/ClementineCttn/r-geospatial-urban/blob/main/episodes/fig/Brielle_NL.jpeg){width=40%}
+
+![Brielle, New Jersey](https://raw.githubusercontent.com/ClementineCttn/r-geospatial-urban/blob/main/episodes/fig/Brielle_NJ.jpeg "Brielle, New Jersey"){width=40%}
+
+
+We should therefore try to be as unambiguous as possible by adding a country code or district name.
+
+
+```{r}
+nominatim_polygon <- nominatimlite::geo_lite_sf(address = "Brielle, NL", points_only = FALSE)
```

Comment:
I did in the first instance of the workshop.
Since then it does not work as it used to, so I switched to nominatim (and also I like to show where it comes from anyway)